### PR TITLE
DROOLS-2798: [DMN Designer] Data-types: Grid: Invocation

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/DROOLS-2798

This PR depends on https://github.com/kiegroup/kie-wb-common/pull/2040 being first merged (as the changes for "Invocation grid" build on top of the changes for "Relation grid"). Therefore review of this PR can be deferred until the other PR has been approved and merged...

### CODE REVIEW

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by karreiro** [comment]
Jenkins please retest this.

(maybe we could rebase these PRs: #2040 #2041 #2042 and #2055).

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
@karreiro I was going to rebase them to single commits as each predecessor is merged.

**Comment by manstis** [comment]
@jomarko This PR is good to review now; thank-you.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/expressions/types/invocation/InvocationGridRendererTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    public void testHeaderDimensionsWhenHeaderIsNotHidden() {
+        final GridRenderer renderer = new InvocationGridRenderer(true);
+
+        assertEquals(InvocationGridRenderer.HEADER_HEIGHT,
+                     renderer.getHeaderHeight(),
+                     0.0);
+        assertEquals(InvocationGridRenderer.HEADER_ROW_HEIGHT,
+                     renderer.getHeaderRowHeight(),
+                     0.0);
+    }
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/expressions/types/invocation/InvocationGridRendererTest.java]: Please explain I would expect 
```java
@Test
    public void testHeaderDimensionsWhenHeaderNotHidden() {
        final GridRenderer renderer = new InvocationGridRenderer(false);

        assertEquals(InvocationGridRenderer.HEADER_HEIGHT,
                     renderer.getHeaderHeight(),
                     0.0);
        assertEquals(InvocationGridRenderer.HEADER_ROW_HEIGHT,
                     renderer.getHeaderRowHeight(),
                     0.0);
    }

    @Test
    public void testHeaderDimensionsWhenHeaderIsHidden() {
        final GridRenderer renderer = new InvocationGridRenderer(true);

        assertEquals(InvocationGridRenderer.HEADER_HEIGHT,
                     0.0,
                     0.0);
        assertEquals(InvocationGridRenderer.HEADER_ROW_HEIGHT,
                     0.0,
                     0.0);
    }
```

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/invocation/InvocationGrid.java [hunk 2]
```diff
@@ -134,15 +143,26 @@ public InvocationUIModelMapper makeUiModelMapper() {
 
     @Override
     protected void initialiseUiColumns() {
+        HasTypeRef hasTypeRef = expression.get();
+        final DMNModelInstrumentedBase base = hasExpression.asDMNModelInstrumentedBase();
+        if (base instanceof HasVariable) {
+            final HasVariable hasVariable = (HasVariable) base;
+            hasTypeRef = hasVariable.getVariable();
+        }
```
[the remark this answers is not in the document]
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/invocation/InvocationGrid.java]: @jomarko Yes, I agree. @karreiro spotted something similar [here](https://github.com/kiegroup/kie-wb-common/pull/2055#discussion_r212084447). I'd prefer to log these things and circle back to correct them all once all the current PRs are merged; if OK?

**Comment by jomarko** [comment]
@manstis  Please react on the last comment and I can approve, manual check passed.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/invocation/InvocationGrid.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/invocation/InvocationGrid.java]: See https://issues.jboss.org/browse/DROOLS-2919

**Pull request merged**